src/components/editor-content/editor-content.tsx

```diff
@@ -52,6 +85,184 @@ const EditorContent: React.FC<Props> = () => {
     );
   }
 
+  /**
+   * Renders selector of filter scope
+   */
+  const renderSelectScope = () => {
+    const handleChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+      setScope(event.target.value as DatePickerView);
+      switch(event.target.value as string) {
+        case "date":
+          setDateFormat("dd/MM/yyyy")
+          break;
+        case "month":
+          setDateFormat("MM/yyyy")
+          break;
+        case "year":
+          setDateFormat("yyyy")
+          break;
+      }
+    };
+
+    return (
+      <>
+      <FormControl variant="outlined" className={ classes.selectScope }>
+        <TextField
+          select
+          id="scope-select-outlined"
+          size="small"
+          value={ scope }
+          onChange={ handleChange }
+        >
+          <MenuItem value={ "week" }>{ strings.editorContent.scopeWeek.toUpperCase() }</MenuItem>
+          <MenuItem value={ "date" }>{ strings.editorContent.scopeDate.toUpperCase() }</MenuItem>
+          <MenuItem value={ "month" }>{ strings.editorContent.scopeMonth.toUpperCase() }</MenuItem>
+          <MenuItem value={ "year" }>{ strings.editorContent.scopeYear.toUpperCase() }</MenuItem>
```

Comment:
Is there any way to avoid using something like vale={ "date" }? How about DatePickerView or an enum?
and for strings if it needs to be used as upper case probably it can be set to upper case in the localization file directly